modules/navier_stokes/include/actions/NSFVAction.h

```diff
@@ -0,0 +1,235 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Action.h"
+
+#include "MooseEnum.h"
+#include "libmesh/fe_type.h"
+
+/**
+ * This class allows us to have a section of the input file like the
+ * following which automatically adds variables, kernels, aux kernels, bcs
+ * for setting up the incompressible/weakly-compressible Navier-Stokes equations.
+ * Create it using the following input syntax:
+ * [Modeles]
+ *   [NavierStokesFV]
+ *     param_1 = value_1
+ *     param_2 = value_2
+ *     ...
+ *   []
+ * []
+ */
+class NSFVAction : public Action
+{
+public:
+  static InputParameters validParams();
+
+  NSFVAction(InputParameters parameters);
+
+  virtual void act() override;
+
+protected:
+  /// Type that we use in Actions for declaring coupling
+  typedef std::vector<VariableName> CoupledName;
+
+  /// Functions for defining the variables depending on the compressibility option
+  /// selected by the user
+  void addINSVariables();
+
+  /// Function which adds the RhieChow interpolator user objects
+  void addRhieChowUserObjects();
+
+  /// Functions that add the initial conditions for the variables
+  void addINSInitialConditions();
+
+  /// Fnction adding kernels for the incompressible continuity equation
+  void addINSMassKernels();
+
+  /**
+   * Functions adding kernels for the incompressible momentum equation
+   * If the material properties are not constant, these can be used for
+   * weakly-compressible simulations as well.
+   */
+  void addINSMomentumTimeKernels();
+  void addINSMomentumViscousDiscipationKernels();
+  void addINSMomentumMixingLengthKernels();
+  void addINSMomentumAdvectionKernels();
+  void addINSMomentumPressureKernels();
+  void addINSMomentumGravityKernels();
+  void addINSMomentumBoussinesqKernels();
+  void addINSMomentumFrictionKernels();
+
+  /**
+   * Functions adding kernels for the incompressible energy equation
+   * If the material properties are not constant, some of these can be used for
+   * weakly-compressible simulations as well.
+   */
+  void addINSEnergyTimeKernels();
+  void addINSEnergyHeatConductionKernels();
+  void addINSEnergyAdvectionKernels();
+  void addINSEnergyAmbientConvection();
+  void addINSEnergyExternalHeatSource();
+
```

Comment:
need to add scalar kernels too

Reply:
Added passive scalar transport kernels too!